mlrun/runtimes/mounts.py

```diff
@@ -0,0 +1,574 @@
+    user: str = "",
+    secret: typing.Optional[str] = None,
+    volume_mounts: typing.Optional[list[VolumeMount]] = None,
+) -> typing.Callable[["KubeResource"], "KubeResource"]:
+    """Modifier function to apply to a Container Op to volume mount a v3io path
+
+    :param name: the volume name
+    :param remote: the v3io path to use for the volume (~/ prefix will be replaced with /users/<username>/)
+    :param access_key: the access key used to auth against v3io (default: V3IO_ACCESS_KEY env var)
+    :param user: the username used to auth against v3io (default: V3IO_USERNAME env var)
+    :param secret: k8s secret name for the username and access key
+    :param volume_mounts: list of VolumeMount; if empty, defaults to mounting /v3io and /User
+    """
+    volume_mounts, user = _enrich_and_validate_v3io_mounts(
+        remote=remote,
+        volume_mounts=volume_mounts,
+        user=user,
+    )
+
+    def _attach_volume_mounts_and_creds(runtime: "KubeResource"):
+        vol = v3io_to_vol(name, remote, access_key, user, secret=secret)
+        runtime.spec.with_volumes(vol)
+
+        for volume_mount in volume_mounts:
+            runtime.spec.with_volume_mounts(
+                {
+                    "mountPath": volume_mount.path,
+                    "name": name,
+                    "subPath": volume_mount.sub_path,
+                }
+            )
+
+        if not secret:
+            runtime = v3io_cred(access_key=access_key, user=user)(runtime)
+        return runtime
+
+    return _attach_volume_mounts_and_creds
+
+
+def mount_spark_conf() -> typing.Callable[["KubeResource"], "KubeResource"]:
+    """Modifier function to mount Spark configuration."""
+
+    def _mount_spark(runtime: "KubeResource"):
+        runtime.spec.with_volume_mounts(
+            {
+                "mountPath": "/etc/config/spark",
+                "name": "spark-master-config",
+            }
+        )
+        return runtime
+
+    return _mount_spark
+
+
+def mount_v3iod(
+    namespace: str, v3io_config_configmap: str
+) -> typing.Callable[["KubeResource"], "KubeResource"]:
+    """Modifier function to mount v3iod configuration."""
+
+    def _mount_v3iod(runtime: "KubeResource"):
+        def add_vol(name, mount_path, host_path):
+            runtime.spec.with_volumes(
+                {
+                    "name": name,
+                    "hostPath": {
+                        "path": host_path,
+                        "type": "",
+                    },
+                }
+            )
+            runtime.spec.with_volume_mounts(
+                {
+                    "mountPath": mount_path,
+                    "name": name,
+                }
+            )
+
+        # this is a legacy path for the daemon shared memory
+        host_path = "/dev/shm/"
+
+        # path to shared memory for daemon was changed in Iguazio 3.2.3-b1
+        igz_version = mlconf.get_parsed_igz_version()
+        if igz_version:
```

Comment:
actually all of this is not needed. we can assume iguazio and iguazio version >= 3.5 here